ncwriter/schema.py

```diff
@@ -18,28 +17,53 @@
 # Create a new validator class (based on Draft4Validator) to allow templates to use
 # * Python types or numpy dtypes to specify variable data types; and
 # * numpy arrays to specify variable data.
-meta = deepcopy(Draft4Validator.META_SCHEMA)
-meta['definitions']['simpleTypes']['enum'].append('datatype')
-types = deepcopy(Draft4Validator.DEFAULT_TYPES)
-types['datatype'] = (basestring, type, np.dtype)
-types['array'] = (list, np.ndarray)
-TemplateValidator = validators.create(meta_schema=meta,
-                                      validators=Draft4Validator.VALIDATORS,
-                                      default_types=types
-                                      )
+
+TemplateValidator = validators.create(meta_schema=Draft4Validator.META_SCHEMA,
+                                      validators=Draft4Validator.VALIDATORS)
+format_checker = FormatChecker()
+
+TYPE_STRINGS = ('int', 'int8', 'int16', 'int32', 'int64',
+                'float', 'float8', 'float16', 'float32', 'float64', 'double',
+                'S1', 'c', 'i1', 'b', 'B', 'u1', 'i2', 'h', 's', 'u2', 'i4', 'i', 'l', 'u4',
+                'i8', 'u8', 'f4', 'f', 'f8', 'd'
+                )
+
+
+@format_checker.checks('datatype')
+def is_datatype(value):
+    """Return whether the given value is a valid data type specification for a NetCDF variable"""
+    if isinstance(value, basestring):
+        if value in TYPE_STRINGS:
```

Comment:
Already a boolean expression, so can just be:
```python
return value in TYPE_STRINGS
```

Reply:
:+1: Ha, of course! That's the sort of thing I've pointed out before when reviewing other people's code, but it's different when you're writing it yourself...

Actually I'm thinking of replacing the string checking here with an `enum` type in the schema